nautobot_circuit_maintenance/handle_notifications/handler.py

```diff
@@ -164,6 +164,18 @@ def create_or_update_circuit_maintenance(
     maintenance_id = f"{raw_entry.provider.slug}-{parser_maintenance.maintenance_id}"
     try:
         circuit_maintenance_entry = CircuitMaintenance.objects.get(name=maintenance_id)
+        last_parsed_notification = (
+            circuit_maintenance_entry.parsednotification_set.order_by("raw_notification__date").reverse().last()
+        )
+
+        # If the notification is older than the latest one used to update the CircuitMaintenance, we skip updating it
+        if last_parsed_notification and last_parsed_notification.date > datetime.datetime.fromtimestamp(
+            parser_maintenance.stamp, tz=datetime.timezone.utc
+        ):
+            if logger.debug:
+                logger.log_debug(f"Not updating CircuitMaintenance {maintenance_id} because the notification is older.")
```

Comment:
Might be good to add more specifics to the log message, something like:

```suggestion
        parser_maintenance_datetime = datetime.datetime.fromtimestamp(
            parser_maintenance.stamp, tz=datetime.timezone.utc
        )
        if last_parsed_notification and last_parsed_notification.date > parser_maintenance_datetime:
            if logger.debug:
                logger.log_debug(
                    f"Not updating CircuitMaintenance {maintenance_id} because the notification is from "
                    f"{parser_maintenance_datetime}, older than the most recent notification from {last_parsed_notification.date}."
                )
```